src/api/endpoints/logged_activities.py

```diff
@@ -158,7 +160,11 @@ def put(cls, logged_activity_id=None):
                 return response_builder(dict(
                     message='Not allowed. Activity is already in pre-approval.'
                 ), 401)
+            if not result.get('activity_type_id'):
+                result['activity_type_id'] = logged_activity.activity_type_id
 
+            date = payload.get('date')
```

Comment:
Woops! Seems like I missed something here, you're getting the date directly from the payload instead of the validated marshmallow result. Should be `result.get('date')` instead. After which, the line below (162) should first check if the `date` is none before doing `result['date'] = logged_activity.activity_date`. 

If you do that, line 179-184 can then simply be replaced by `logged_activity.activity_date = parsed_result.activity_date`
Let me know if this is still a bit unclear.